contracts/access/Bouncer.sol

```diff
@@ -146,14 +160,27 @@ contract SignatureBouncer is Ownable, RBAC {
    * and then recover the signature and check it against the bouncer role
    * @return bool
    */
-  function isValidDataHash(bytes32 hash, bytes _sig)
+  function isValidDataHash(bytes32 _hash, bytes _sig)
     internal
     view
     returns (bool)
   {
-    address signer = hash
+    // if the sender is a delegate AND supports isValidSignature, we delegate validation to them
+    // this allows someone who wants custom validation logic (perhaps they check another contract's state)
+    // to code their own delegate. The users then submit actions to the delegate contract, which proxies
+    // them to this contract, which then pings-back to check if it's cool.
+    // This is also useful for contracts-as-identities: your identity contract implements isValidSignature
+    // and can then recover the signer and check it against the whitelisted ACTION keys.
+    if (hasRole(msg.sender, ROLE_DELEGATE) &&
+        msg.sender.supportsInterface(InterfaceId_BouncerDelegate)) {
+      return IBouncerDelegate(msg.sender).isValidSignature(_hash, _sig);
```

Comment:
It is important to note that `msg.sender` might not be the delegated contract address. 

Take for example a token sale contract that utilizes bouncer functionalities to have an "off-chain" whitelist. The "signer" (the `owner`), who whitelists buyers address, is a multisig contract with a threshold of 3 signatures (out of 5). Hence, every buyer now has 3 signatures they would need to pass to the tokensale contract to be able to participate in the sale. In this case, the buyer is `msg.sender`, the "validator" is the multisig contract (e.g. `ROLE_BOUNCER == WHITELIST_SIGNER`). and the signers are one of the 5 team members. 

Hence, I believe one need to pass the `delegate` / `bouncer` address which contains the validation logic. `

Reply:
I see what you mean, yeah; so we should allow a `_delegate` as part of the calldata, which is then signed and used in this function. Cool, will make that change